src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+import org.slf4j.LoggerFactory;
+
+import se.simonsoft.cms.item.CmsItem;
+import se.simonsoft.cms.item.CmsItemId;
+import se.simonsoft.cms.item.CmsItemLockCollection;
+import se.simonsoft.cms.item.CmsItemPath;
+import se.simonsoft.cms.item.CmsRepository;
+import se.simonsoft.cms.item.RepoRevision;
+import se.simonsoft.cms.item.commit.CmsCommit;
+import se.simonsoft.cms.item.commit.CmsPatchItem;
+import se.simonsoft.cms.item.commit.CmsPatchset;
+import se.simonsoft.cms.item.commit.FileAdd;
+import se.simonsoft.cms.item.commit.FileModificationLocked;
+import se.simonsoft.cms.item.commit.FolderExist;
+import se.simonsoft.cms.item.impl.CmsItemIdArg;
+import se.simonsoft.cms.item.info.CmsItemLookup;
+import se.simonsoft.cms.item.info.CmsItemNotFoundException;
+import se.simonsoft.cms.item.info.CmsRepositoryLookup;
+import se.simonsoft.cms.item.properties.CmsItemPropertiesMap;
+import se.simonsoft.cms.transform.config.databind.TransformConfig;
+import se.simonsoft.cms.xmlsource.handler.XmlSourceReader;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceDocumentS9api;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceReaderS9api;
+import se.simonsoft.cms.xmlsource.transform.SaxonOutputURIResolverXdm;
+import se.simonsoft.cms.xmlsource.transform.TransformOptions;
+import se.simonsoft.cms.xmlsource.transform.TransformStreamProvider;
+import se.simonsoft.cms.xmlsource.transform.TransformerService;
+import se.simonsoft.cms.xmlsource.transform.TransformerServiceFactory;
+
+public class TransformServiceXsl implements TransformService {
+	
+	private final CmsCommit commit;
+	private final CmsItemLookup itemLookup;
+	private final TransformerServiceFactory transformerServiceFactory;
+	private final CmsRepositoryLookup repoLookup;
+	private final Map<String, TransformerService> stylesheets;
+	private final XmlSourceReaderS9api sourceReader;
+	private final TransformerService transformerIdentity;
+	
+	private static final String TRANSFORM_LOCK_COMMENT = "Locked for transform";
+	private static final String TRANSFORM_BASE_PROP_KEY = "abx:TransformBase";
+	private static final String TRANSFORM_NAME_PROP_KEY = "abx:TransformName";
+	private static final Logger logger = LoggerFactory.getLogger(TransformServiceXsl.class);
+
+	@Inject
+	public TransformServiceXsl(
+			CmsCommit commit,
+			CmsItemLookup itemLookup,
+			CmsRepositoryLookup lookupRepo,
+			TransformerServiceFactory transfromerServiceFactory,
+			Map<String, TransformerService> stylesheets,
+			XmlSourceReader sourceReader
+			) {
+		
+		this.commit = commit;
+		this.itemLookup = itemLookup;
+		this.repoLookup = lookupRepo;
+		this.transformerServiceFactory = transfromerServiceFactory;
+		this.transformerIdentity = transfromerServiceFactory.buildTransformerService(new StreamSource(this.getClass().getClassLoader().getResourceAsStream("se/simonsoft/cms/xmlsource/transform/identity.xsl")));
+		this.stylesheets = stylesheets;
+		if (sourceReader instanceof XmlSourceReaderS9api) {
```

Comment:
Is this needed?

Reply:
Followed pattern in cms-author and cms-xmlsource. To be honest I'm not sure if needed.

Reply:
It is not needed when injecting XmlSourceReaderS9api specifically. See more modern code.